src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -559,6 +559,10 @@ public void GetNameAndArityOfSimpleName(SyntaxNode node, out string name, out in
             }
         }
 
+        public bool LooksGeneric(SyntaxNode simpleName)
+            => simpleName.IsKind(SyntaxKind.GenericName) ||
+               simpleName.GetLastToken().GetNextToken().Kind() == SyntaxKind.LessThanToken;
```

Comment:
I think it seems unlikely for this to occur, and also unlikely for `foo` to be generic if it does occur. So it could go either way.